samples/batch/transcription-enabled-storage/Connector/BatchClient.cs

```diff
@@ -7,67 +7,67 @@ namespace Connector
 
-        private readonly string HostName;
+        private static readonly TimeSpan DefaultTimeout = TimeSpan.FromMinutes(2);
 
-        private ILogger Log;
+        private static readonly TimeSpan GetFilesTimeout = TimeSpan.FromMinutes(5);
 
-        public BatchClient(string subscriptionKey, string hostName, ILogger log)
-        {
-            SubscriptionKey = subscriptionKey;
-            HostName = hostName;
-            Log = log;
-        }
+        private static HttpClient HttpClient = new HttpClient() { Timeout = System.Threading.Timeout.InfiniteTimeSpan };
 
         public static bool IsThrottledOrTimeoutStatusCode(HttpStatusCode statusCode)
         {
             if (statusCode == HttpStatusCode.TooManyRequests ||
                 statusCode == HttpStatusCode.GatewayTimeout ||
                 statusCode == HttpStatusCode.RequestTimeout ||
-                statusCode == HttpStatusCode.BadGateway ||
-                statusCode == HttpStatusCode.NotFound)
+                statusCode == HttpStatusCode.BadGateway)
             {
                 return true;
             }
 
             return false;
         }
 
-        public Task<IEnumerable<Transcription>> GetTranscriptionsAsync()
+        public static Task<TranscriptionReportFile> GetTranscriptionReportFileFromSasAsync(string sasUri, ILogger log)
+        {
+            return GetAsync<TranscriptionReportFile>(sasUri, null, DefaultTimeout, log);
+        }
+
+        public static Task<SpeechTranscript> GetSpeechTranscriptFromSasAsync(string sasUri, ILogger log)
         {
-            var path = $"{HostName}{SpeechToTextBasePath}Transcriptions";
-            return this.GetAsync<IEnumerable<Transcription>>(path);
+            return GetAsync<SpeechTranscript>(sasUri, null, DefaultTimeout, log);
         }
 
-        public Task<Transcription> GetTranscriptionAsync(Guid id)
+        public static Task<Transcription> GetTranscriptionAsync(string transcriptionLocation, string subscriptionKey, ILogger log)
         {
-            var path = $"{HostName}{SpeechToTextBasePath}Transcriptions/{id}";
-            return this.GetAsync<Transcription>(path);
+            return GetAsync<Transcription>(transcriptionLocation, subscriptionKey, DefaultTimeout, log);
         }
 
-        public async Task<TranscriptionFiles> GetTranscriptionFilesAsync(Guid id)
+        public static async Task<TranscriptionFiles> GetTranscriptionFilesAsync(string transcriptionLocation, string subscriptionKey, ILogger log)
         {
-            var path = $"{HostName}{SpeechToTextBasePath}Transcriptions/{id}/files";
+            var path = $"{transcriptionLocation}/files";
 
             var combinedTranscriptionFiles = new List<TranscriptionFile>();
 
             do
             {
-                var transcriptionFiles = await this.GetAsync<TranscriptionFiles>(path).ConfigureAwait(false);
+                // Delay to avoid throttling
+                await Task.Delay(2000).ConfigureAwait(false);
```

Comment:
do you need exponential backoff, or at least  increase the delay time based on retry times.

Reply:
ok, this is just wait for retrieving the nextlink. Is this really needed? I would avoid to use hardcoded delay here, since it just slow the processing. What we should do is adding retry in http calls in case of Status code is 429 or something else. You can use polly like CRIS backend, or manually do that.

Reply:
I just thought that since I'm sending multiple requests one after another here against the service, I'm trying to avoid getting throttled. 
I think I can just remove it, because if I'm doing a GET Files, I'm either getting a fast response if there are only a few files - or a slow response for many files. And if I get a slow response, throttling is probably not a problem anyway.